components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -352,6 +360,79 @@ protected String getUserStoreAppendedName(String userName) {
         return userName;
     }
 
+    protected org.wso2.carbon.user.core.common.User getUser(AuthenticatedUser authenticatedUser)
+            throws AuthenticationFailedException {
+
+        org.wso2.carbon.user.core.common.User user = null;
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        if (tenantDomain == null) {
+            return null;
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        try {
+            UserRealm userRealm = FrameworkServiceDataHolder.getInstance().getRealmService()
+                    .getTenantUserRealm(tenantId);
+            if (userRealm != null) {
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+                List<org.wso2.carbon.user.core.common.User> userList
+                        = ((AbstractUserStoreManager) userStoreManager).getUserListWithID(
+                        USERNAME_CLAIM, authenticatedUser.getUserName(), null);
+                if (userList.isEmpty()) {
+                    userList = ((AbstractUserStoreManager) userStoreManager).getUserListWithID(
+                            EMAIL_ADDRESS_CLAIM, authenticatedUser.getUserName(), null);
+                }
+                userList = getValidUsers(userList);
+                if (CollectionUtils.isEmpty(userList)) {
+                    return null;
+                }
+                if (userList.size() > 1) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("There are more than one user with the provided username claim value: "
+                                + authenticatedUser.getUserName());
+                    }
+                    return null;
+                }
+                user = userList.get(0);
+            } else {
+                log.error("Cannot find the user realm for the given tenant: " + tenantDomain);
+            }
+        } catch (UserStoreException e) {
+            String msg = "getUserListWithID function failed while retrieving the user list.";
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
```

Comment:
since we are already throwing the error, it will get handled in the place where we are calling this method hence no need to print it as debug log here. Change the explanation with some more meaningful message, no need to specify the method name.

Reply:
Fixed with 3299151e890857486827301005d986c6fdc7bb90